### ISSUE TITLE
Palette Search - Enter does not activate clear button

### ISSUE THREAD

**Issue by jmwmusic** [musescore/MuseScore#33897]
### Issue type

Accessibility issue (e.g. for keyboard-only or screen reader users)

### Description with steps to reproduce

In the Palette search area the clear button can only be activated with space not enter.  Buttons should be able to be activated with enter.

STEPS:
1.  Go to Palettes and press enter on the search button.
2. Type some text in the edit field.
3. Press down arrow to navigate to the Clear button.
4. Press Enter.
5. Notice that either nothing happens, or on 1 occasion focus returned to the score.
6. Navigate again to the clear button and press space.  This works and the text box is now empty.

### Supporting files, videos and screenshots

N/A

### In which versions of MuseScore Studio is this issue present?

4.7.3

### Regression

I was unable to check

### Operating system

Windows 11 with NVDA

### Checklist

- [x] This report follows the [guidelines](https://github.com/musescore/MuseScore/wiki/Reporting-bugs-and-issues) for reporting bugs and issues
- [x] I have verified that this issue has not been logged before, by searching the [issue tracker](https://github.com/musescore/MuseScore/issues) for similar issues
- [x] I have attached all requested files and information to this report
- [x] I have attempted to identify the root problem as concisely as possible, and have used minimal reproducible examples where possible
